rust/main/chains/hyperlane-cosmos/src/types.rs

```diff
@@ -30,6 +31,21 @@ impl From<IsmType> for ModuleType {
     }
 }
 
+pub trait ToCosmosHttpUrl {
+    fn to_cosmos_http_url(
+        &self,
+    ) -> Result<tendermint_rpc::HttpClientUrl, tendermint_rpc::error::Error>;
+}
+
+impl ToCosmosHttpUrl for url::Url {
+    fn to_cosmos_http_url(
+        &self,
+    ) -> Result<tendermint_rpc::HttpClientUrl, tendermint_rpc::error::Error> {
+        let tendermint_url = tendermint_rpc::Url::try_from(self.to_owned())?;
+        tendermint_rpc::HttpClientUrl::try_from(tendermint_url)
+    }
```

Comment:
since this is only used once, what's the advantage to defining it as a trait vs using it inline in `CosmosRpcClient:new`? Simpler is usually better so would prefer it inline

Reply:
Wanted to reduce the bloat from the `new` function call. But fair point, moved inline to `CosmosRpcClient::new` ✅